tests/test_cfg/test_cfg_generator.py

```diff
@@ -2,60 +2,55 @@
 Test suite for generating control flow graphs using the PythonTA API.
 """
 import os.path
-import tempfile
-from typing import Tuple
 
 import pytest
 
 import python_ta.cfg as cfg
 
-TEST_SCRIPT = """
-def foo() -> None:
-    for i in range(1, 10):
-        if i < 5:
-            print("hi")
-"""
-
 
 @pytest.fixture(autouse=True)
 def reset_dir(request, monkeypatch):
     """Reset the current working directory after each test."""
     monkeypatch.chdir(request.fspath.dirname)
 
 
-def create_script(code: str) -> Tuple[str, str, str]:
-    """Returns a 3-tuple containing the name of the temporary file corresponding to the Python file
-    of script, and the names of the two files that will be produced by graphviz."""
-    script = tempfile.NamedTemporaryFile(suffix=".py", mode="w+t", delete=False)
-    script.writelines(code)
-    script.flush()
-    script.close()
+def test_script_external_call() -> None:
+    """Test that generate_cfg correctly creates both graph files when the script does not contain
+    the call to create its control flow graph."""
+    script_name = "file_fixtures/my_file.py"
+    dot_file_path = os.path.splitext(os.path.basename(script_name))[0]
+    svg_file_path = dot_file_path + ".svg"
 
-    # Get the paths of the files of interest (temp file + graphviz files that will be created)
-    dir_name = os.path.dirname(script.name)
-    file_path = os.path.splitext(script.name)[0]
-    svg_file_path = file_path + ".svg"
+    # Create the graphviz files using my_file.py
+    cfg.generate_cfg(mod=script_name, auto_open=False)
 
-    # Change directory to the temporary directory so the graphviz files will be created there
-    os.chdir(dir_name)
+    # Check if the graphviz files were created
+    assert os.path.isfile(dot_file_path) and os.path.isfile(svg_file_path)
 
-    return script.name, file_path, svg_file_path
+    # Teardown: remove the graphviz generated files
+    os.remove(dot_file_path)
```

Comment:
It's not ideal to put cleanup code at the end of a test body, since (for example) if an earlier assertion fails, the code won't run. Instead this is a good time to use a `yield` fixture: see https://docs.pytest.org/en/6.2.x/fixture.html#teardown-cleanup-aka-fixture-finalization.

*Note*: the easiest way to use such a fixture is to move the logic of the specific filenames (`script_name`, `dot_file_path`, `svg_file_path`) into the fixture and pass them into the test case. This will also remove some code duplication across your two tests.